core/src/main/scala/kafka/cluster/Replica.scala

```diff
@@ -167,7 +167,10 @@ class Replica(val brokerId: Int,
 
   def convertHWToLocalOffsetMetadata() = {
     if (isLocal) {
-      highWatermarkMetadata = log.get.convertToOffsetMetadata(highWatermarkMetadata.messageOffset)
+      highWatermarkMetadata = log.get.convertToOffsetMetadata(highWatermarkMetadata.messageOffset).getOrElse {
```

Comment:
Would it be useful to have a short comment explaining when we can hit this case?

Reply:
Sure. I added the following comment:

```
Convert hw to local offset metadata by reading the log at the hw offset. 
If the hw offset is out of range, return the first offset of the first log segment as the offset metadata.
```